packages/compass-e2e-tests/helpers/commands/wait-for-element.js

```diff
@@ -0,0 +1,57 @@
+const { delay } = require('../delay');
+
+module.exports = function (app) {
+  return async function waitForElement(
+    selector,
+    {
+      timeout = 5000,
+      interval = 200,
+      mustExist = true,
+      mustBeVisible = true,
+      mustNotTransition = true,
+      returnElement = false,
+      existError,
+      visibleError,
```

Comment:
This feels a bit overloaded with options, maybe it's worth splitting in multiple commands with their own purpose? Another thing that caught my eye is even though I generally like the options object pattern, this goes against the webdriver API patterns and now we have a mix of two patterns that test writers need to be aware of

Reply:
I deliberately made it one command so I could collapse a dozen lines into one. I wouldn't mind wrappers if you want ones for the common cases, though.

Reply:
My thinking is that before you can interact with a thing you always have to wait for it to exist and become visible and stop transitioning and checking those things individually (like you've done in the commands) allows the error to tell you exactly which part went wrong. Without a command that checks all the common stuff we're going to have walls of boilerplate everywhere and we're also going to have the thing happen where in some cases we check only some of it, in other cases the rest.. We're also going to have deeply nested code, etc.

Reply:
I'd also caution against using the webdriver commands directly as they are pretty low level building blocks. I'd like common things like "wait for a modal" to be one command and "wait for a modal and dismiss it" to be two and "wait for the confirmation message and close it" to be one or two. Otherwise I think we're going to have a proliferation of commands that do these things.

Reply:
@gribnoysup Actually let's do a call about this sometime. I'm now second-guessing what I wrote and I coded and I think we can come up with something we'll both like. Let's just discuss the problems to solve. If any ;)

Reply:
I couldn't really tell what webdriver's pattern is:

https://webdriver.io/docs/api/browser/waitUntil/ (has condition then options hash)
vs
http://v4.webdriver.io/api/utility/waitForVisible.html (has positional params)
vs
http://v4.webdriver.io/api/state/isVisible.html (which has neither)

Reply:
OK so it depends on where you look: http://v4.webdriver.io/api/utility/waitUntil.html

Reply:
We are using wdio v4 because this is what spectron is bundled with and it looks like it's pretty consistent there, right? It's a fair point though as we will have to upgrade spectron when updating electron versions and this will kinda switch us from one API style to another eventually

Reply:
I think both must be supported because I switched one thing you called with positional arguments over to an options hash and it still worked.